.golangci.yml

```
@@ -17,12 +16,10 @@ linters:
     - nakedret
     - revive
     - staticcheck
-    - structcheck
     - stylecheck
     - unconvert
     - unparam
     - unused
-    - varcheck
 run:
   go: '1.17'
```

I know is not part of this PR but, should this be modified?
```suggestion
  go: '1.18'
```

---

@iblancasa the project went to go 1.18 early.   In those early days many of the linters broke.  there were negative consequences of running as 1.18 (as well as some for running as 1.17).   Running as 1.17 made the most sense at the time... but it is definitely worth revisiting that now.   thanks for calling it out!   I will review now to see the consequences.

---

switching to running 1.18 linters results in the following disable linters.   I'm surprised this is still the case.

```
WARN [linters context] gosimple is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
WARN [linters context] staticcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
WARN [linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
WARN [linters context] stylecheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
WARN [linters context] unparam is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
WARN [linters context] unused is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
```

---

Interesting. I think we can leave it there.

Thanks!

---

@iblancasa the logic behind stay with go1.17 running of linters is we can't write and have 1.18 style code linted (so no generics) but we get all the 1.17 linting.